### ISSUE TITLE
Limit/throttle number of packages that can be registered at the same time by a user

### ISSUE THREAD

**Issue by giordano** [JuliaRegistries/RegistryCI.jl#350]
Prompted by [a new attack to PyPi](https://twitter.com/DataNerdery/status/1366263351685226499), it was suggested on Slack that we may want to limit the number of new registrations a user can submit at the same time.  Sascha suggested

> Maybe a throttle could be useful to have (in general, not specifically to mitigate this)? For example if the new registration activity explodes, or if a user registers more than y packages in z time, automerge will wait a bit longer?

while Stefan proposed

> Max number of outstanding registrations before more can be made?

Of course these controls can always be circumvented, but all of this + 3-day waiting period should discourage possible attackers.

I'm not sure this actually requires any help from `RegistryCI.jl`, in particular Stefan's suggestion may be easy to implement just using GitHub Actions API in [General](https://github.com/JuliaRegistries/General), but I figured I'd open an issue here to float the idea.

**Comment by SaschaMann** [comment]
Stefan's suggestion could be implemented through GitHub's Search API with a query `state:open type:pr label:"new package" author:$USERNAME` and perhaps `per_page:$LIMIT`.

Perhaps AutoMerge could cache the results for each author to reduce the number of required API calls.

---

This isn't a direct mitigation to the attack on PyPi but it would help slow down flooding-based attacks so that registry maintainers have enough time to react to it.

**Comment by giordano** [comment]
> This isn't a direct mitigation to the attack on PyPi but it would help slow down flooding-based attacks so that registry maintainers have enough time to react to it.

Yes, I also forgot to mention that these checks are only meant to slow down flooding-based attacks.  If a registry maintainer recognises a package developer is simply very productive and in good faith, then the pull requests can be manually merged after the regular waiting period.
